openbas-api/src/main/java/io/openbas/rest/dashboard/DashboardApi.java

```diff
@@ -0,0 +1,83 @@
+package io.openbas.rest.dashboard;
+
+import static io.openbas.config.SessionHelper.currentUser;
+import static io.openbas.database.model.User.ROLE_USER;
+import static io.openbas.engine.api.HistogramWidget.HistogramConfigMode.STRUCTURAL;
+import static io.openbas.engine.api.HistogramWidget.HistogramConfigMode.TEMPORAL;
+
+import io.openbas.database.model.Filters;
+import io.openbas.database.model.Widget;
+import io.openbas.database.raw.RawUserAuth;
+import io.openbas.database.repository.UserRepository;
+import io.openbas.engine.api.*;
+import io.openbas.engine.model.*;
+import io.openbas.engine.query.EsSeries;
+import io.openbas.rest.custom_dashboard.WidgetService;
+import io.openbas.rest.helper.RestBehavior;
+import io.openbas.service.EsService;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Secured(ROLE_USER)
+@RequiredArgsConstructor
+public class DashboardApi extends RestBehavior {
+
+  public static final String DASHBOARD_URI = "/api/dashboards";
+
+  private final EsService esService;
+  private final UserRepository userRepository;
+  private final WidgetService widgetService;
+
+  @GetMapping(DASHBOARD_URI + "/count/{type}")
+  public long count(@PathVariable String type) {
+    Filters.FilterGroup filterGroup = new Filters.FilterGroup();
+    Filters.Filter filter = new Filters.Filter();
+    filter.setKey("base_entity");
+    filter.setOperator(Filters.FilterOperator.eq);
+    filter.setValues(List.of(type));
+    filterGroup.setFilters(List.of(filter));
+    CountConfig config = new CountConfig("Series01", filterGroup);
+    CountRuntime runtime = new CountRuntime(config);
+    RawUserAuth userWithAuth = userRepository.getUserWithAuth(currentUser().getId());
+    return esService.count(userWithAuth, runtime);
+  }
+
+  @GetMapping(DASHBOARD_URI + "/series/{widgetId}")
+  public List<EsSeries> series(@PathVariable final String widgetId) {
+    Widget widget = this.widgetService.widget(widgetId);
+    if (TEMPORAL.equals(widget.getHistogramWidget().getMode())) {
```

Comment:
Shouldn't we test the type first with something like :
if(Widget.WidgetType.VERTICAL_BAR_CHART.equals(widget.getType())) {

Reply:
I am not sure to understand your point

Reply:
We're doing widget.getHistogramWidget() but what if this widget is not of type histogram ? Shouldn't we test the type of the widget first ? Or maybe I'm missing something ?

Reply:
Oh I see.
The widget type is not correlated with the widget mode.
You can have a vertical bar chart for a temporal or a structured mode.
Temporal: finding over time
Structured: finding by asset

Reply:
Ok, I get it, thanks for the clarification :)